keras/layers/core.py

```diff
@@ -995,6 +995,86 @@ def get_config(self):
         return dict(list(base_config.items()) + list(config.items()))
 
 
+class TimeDistributed(MaskedLayer):
+    def __init__(self, layer, input_shape=None, input_dim=None, input_length=None, weights=None, **kwargs):
+        self.layer = layer
+
+        self.initial_weights = weights
+
+        self.input_dim = input_dim
+        self.input_length = input_length
+
+        if hasattr(self.layer, 'input_ndim'):
+            self.input_ndim = self.layer.input_ndim + 1
+
+        if input_shape:
+            self.set_input_shape((None, ) + input_shape)
+
+        if self.input_dim:
+            kwargs['input_shape'] = (self.input_length, self.input_dim)
+
+        super(TimeDistributed, self).__init__(**kwargs)
+
+    def set_previous(self, layer):
+        self.input_ndim = len(layer.output_shape)
+        super(TimeDistributed, self).set_previous(layer)
+
+    def build(self):
+        try:
+            self.input_ndim = len(self.previous.input_shape)
+        except AttributeError:
+            self.input_ndim = len(self.input_shape)
+
+        self.layer.set_input_shape((None, ) + self.input_shape[2:])
+
+        if hasattr(self.layer, 'regularizers'):
+            self.regularizers = self.layer.regularizers
+
+        if hasattr(self.layer, 'constraints'):
+            self.constraints = self.layer.constraints
+        
+        if hasattr(self.layer, 'trainable_weights'):
+            self.trainable_weights = self.layer.trainable_weights
+
+            if self.initial_weights is not None:
+                self.layer.set_weights(self.initial_weights)
+                del self.initial_weights
+
+    @property
+    def output_shape(self):
+        input_shape = self.input_shape
+        return (input_shape[0], input_shape[1]) + self.layer.output_shape[1:]
+
+    def get_output(self, train=False):
+        def format_shape(shape):
+            if K._BACKEND == 'tensorflow':
+                return map(int, shape)
```

Comment:
I don't think it's acceptable to have backend-specific statements if it can be avoided. Also this would fail most of the time because usually the input shape contains `None`.